### PR TITLE
move dataframe commands to nu-cmd-dataframe

### PR DESCRIPTION
**Pull request by stormasm**
All of the dataframe commands ported over with no issues...

### 11 tests are commented out (for now)

So 100 of the original 111 tests are passing with only 11 tests being ignored for now..

As per our conversation in the core team meeting on Wednesday
I took @jntrnr  suggestion and just commented out the tests dealing
with [IntoDatetime](https://github.com/nushell/nushell/blob/main/crates/nu-command/src/conversions/into/mod.rs)

Later on we can move this functionality out of nu-command if we decide it makes sense...

### The following tests were ignored...

```rust
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_day.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_hour.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_minute.rs

modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_month.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_nanosecond.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_ordinal.rs

modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_second.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_week.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_weekday.rs

modified:   crates/nu-cmd-dataframe/src/dataframe/series/date/get_year.rs
modified:   crates/nu-cmd-dataframe/src/dataframe/series/string/strftime.rs
```

### CODE REVIEW

**Review by amtoine — CHANGES_REQUESTED** [review]
i've started by looking at your last commit related to the tests :yum: 

**could we**, instead of commenting them out, which makes them harder to find and reenable, **mark them as `#[ignore]`?**
next to the `#[test]` or `#[cfg(test)]` :wink:

**Review by amtoine — COMMENTED** [review]
this looks great and simple, thanks a lot @stormasm for tackling the refactoring :heart: 

i've got some remarks after looking at your changes:
- :heavy_check_mark: `crates/nu-command/src/dataframe/` is now gone
- :x: an idea about the commenting of tests (see my other review)
- :question: in the main `Cargo.toml`, we have
```toml
[features]
#...
dataframe = ["nu-command/dataframe"]
```
, maybe the path should change? :open_mouth: 
i'm quite confused `cargo build --features dataframe` still works with this path :thinking:

File: crates/nu-cmd-dataframe/Cargo.toml
[the remark this answers is not in the document]
Comment (stormasm) [reply crates/nu-cmd-dataframe/Cargo.toml:1]: @amtoine good catch !  I removed the polars dependencies from nu-command...
and yes I tested and the dependencies now located in Cargo.toml in nu-cmd-dataframe are the minimal dependencies required for dataframes to work...

**Comment by stormasm** [comment]
> i've started by looking at your last commit related to the tests 😋
> 
> **could we**, instead of commenting them out, which makes them harder to find and reenable, **mark them as `#[ignore]`?** next to the `#[test]` or `#[cfg(test)]` 😉

so unfortunately you can only add

```rust
#[ignore]
```

to individual tests and not to the whole 

```rust
#[cfg(test)]
mod test {
```

which is ideally what we would want...
If there were a way to ignore the whole *mod test* or if you know a way to do it 😄 
that would be better --- I just don't think there is a way...

**Comment by stormasm** [comment]
> this looks great and simple, thanks a lot @stormasm for tackling the refactoring ❤️
> 
> i've got some remarks after looking at your changes:
> 
> * ✔️ `crates/nu-command/src/dataframe/` is now gone
> * ❌ an idea about the commenting of tests (see my other review)
> * ❓ in the main `Cargo.toml`, we have
> 
> ```toml
> [features]
> #...
> dataframe = ["nu-command/dataframe"]
> ```
> 
> , maybe the path should change? 😮 i'm quite confused `cargo build --features dataframe` still works with this path 🤔

This was one of the trickiest parts of the refactor ---- but yes this is the magical incantation that is needed for this to work..

File: crates/nu-cmd-dataframe/Cargo.toml
[the remark this answers is not in the document]
Comment (amtoine) [reply crates/nu-cmd-dataframe/Cargo.toml:1]: amazing, then i'm perfectly fines with that part of your changes :blush:

**Comment by amtoine** [comment]
> so unfortunately you can only add
> 
> ```rust
> #[ignore]
> ```
> 
> to individual tests and not to the whole
>
then i'd say adding `#[ignore]` to the 11 individual tests is better :confused: 

i'm not a huge fan of whole blocks of dead code...
at least, with `#[ignore]` we can locate them easily later and we see them with `cargo test` :+1: 
with dead commented code, we can't do either :sob: 

as they are only 11 of them, this looks feasible? :relieved:

**Comment by amtoine** [comment]
> This was one of the trickiest parts of the refactor ---- but yes this is the magical incantation that is needed for this to work..
> 
ooh ok :confused: 
this is quite confusing because `nu-command/dataframe` looks at lot like the now removed `crates/nu-command/src/dataframe/`, doesn't it?
i really thought it would be called `nu-cmd-dataframe` :open_mouth: 

> Getting all of the correct feature pieces in the right place was mainly the only challenging part of the refactor that took me quite a while to figure out... +1
>
i trust you on this :scream: :laughing:

**Comment by stormasm** [comment]
> > so unfortunately you can only add
> > ```rust
> > #[ignore]
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to individual tests and not to the whole
> 
> then i'd say adding `#[ignore]` to the 11 individual tests is better 😕
> 
> i'm not a huge fan of whole blocks of dead code... at least, with `#[ignore]` we can locate them easily later and we see them with `cargo test` 👍 with dead commented code, we can't do either 😭
> 
> as they are only 11 of them, this looks feasible? 😌

agreed on your point with

```rust
#[ignore = "comment why test is being ignored"]
```

but this will not solve the problem because this code is still there

```rust
#[cfg(test)]
mod test {
    use super::super::super::super::super::IntoDatetime;
    use super::super::super::super::test_dataframe::test_dataframe;
    use super::*;
```

and that is where the problem lies..

we need a way to tell the attribute to ignore the whole 

```rust
#[cfg(test)]
mod test {
```

maybe @jntrnr or @fdncred knows a way to do that....

we can either delete the code entirely
which I don't want to do because after this PR lands we can explore possibly move

```rust
IntoDatetime
```

out of nu-command

or once we determine its not worth doing then

then I could come back in and delete those tests

**Comment by amtoine** [comment]
oooh it's an issue with the imports...
ok i get it now :cry: 

then yeah if someone has a better idea :relieved: 


EDIT: a quick and dirty way to do that might be to set a dummy `cfg` that will never happen
e.g. replacing `#[cfg(test)]` with `#[cfg(dummy)] or `#[cfg(fixme)]`?

**Comment by amtoine** [comment]
anytime @stormasm :wink: :relieved: 
this cratification is an important thing to do :blush:

**Comment by kubouch** [comment]
Marked as a breaking change since we won't build with dataframes by default? (Or do we?)

**Comment by stormasm** [comment]
@kubouch we are good....

we were not building dataframes by default prior to my PR...

you had to pass in a feature flag in order to get dataframes...

its also currently documented here as well...

https://www.nushell.sh/book/dataframes.html

**Comment by stormasm** [comment]
When we publish our next release to crates.io we will have to add the crate

```rust
nu-cmd-dataframe
```

to our script...

**Comment by kubouch** [comment]
@stormasm See https://github.com/nushell/nu_scripts/pull/507, I think that should be it?

**Comment by stormasm** [comment]
> @stormasm See [nushell/nu_scripts#507](https://github.com/nushell/nu_scripts/pull/507), I think that should be it?

@kubouch cool !  thanks for adding that...

**Comment by amtoine** [comment]
@stormasm 
can you confirm me that this is a breaking change, as it requires people in need of the `bits` commands to build with `--features dataframe`? :yum:

**Comment by stormasm** [comment]
@amtoine developers needing the bits command will not need dataframes 😄 

```rust
cargo build --features=extra'
```

They will only have to build with the above command which does not include dataframes

**Comment by amtoine** [comment]
woooopsie, bad copy-paste :scream: 
i've changed the comment :wink:

**Comment by fdncred** [comment]
it's not a breaking change. people who wanted dfrs already had to do `--features=dataframes`

**Comment by amtoine** [comment]
> it's not a breaking change. people who wanted dfrs already had to do `--features=dataframes`
>
oooh you're right, it's just a refactor :+1:

**Pull request merged**